Tasks/CondaAuthenticateV0/condaauthenticatemain.ts

```diff
@@ -0,0 +1,319 @@
+import * as path from 'path';
+import * as tl from 'azure-pipelines-task-lib/task';
+import * as fs from 'fs';
+import * as constants from './constants';
+import * as util from 'azure-pipelines-tasks-packaging-common/util';
+import * as pkgLocationUtils from 'azure-pipelines-tasks-packaging-common/locationUtilities';
+import * as url from 'url';
+
+async function main(): Promise<void> {
+    tl.setResourcePath(path.join(__dirname, 'task.json'));
+
+    try {
+        let configCondarc = tl.getInput(constants.CondaAuthenticateTaskInput.ConfigFile);
+        if (!configCondarc) {
+            throw new Error(tl.loc('ConfigCondarcDoesNotExist', configCondarc));
+        } else if (!configCondarc.endsWith('.condarc')) {
+            throw new Error(tl.loc('CondarcNotCondarc', configCondarc));
+        } else {
+            console.log(tl.loc("AuthenticatingThisConfigCondarc", configCondarc));
+        }
+
+        const condarcFile = fs.readFileSync(configCondarc, 'utf8');
```

Comment:
Separate out condarc functionality into a different file in the same directory for better readability. This file should contain the core authenticating mechanism similar to other protocols.